### PR TITLE
Migrate to Ansible collection

### LINKED ISSUE #613: Epic: Ansible collection compatible repository layout
Thanks a lot for super useful collection of ansible roles and playbooks! ❤️ But to be honest, I am sick and tired of using the pesky and partially broken Git submodules. 😡 To mend this, I hereby suggest this epic issue of making this repository compatible with the [Ansible collection repository layout](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html).

Apart from being a big step towards resolving https://github.com/confluentinc/cp-ansible/issues/113, I see the following benefits of doing this:
- Enforcing a standard directory layout will make it more obvious where new things should be added.
- The `ansible-galaxy` CLI also supports Git repository references (branch, tag, commit) to install collections (and roles). This will make it a lot easier to install cp-ansible and use it as a part of a bigger Ansible setup (as we do with submodule now).

I had look on what would need to be done, and I noted this (please add if I missed something):

- [x] Add `galaxy.yml` to the repository root. The content of this file will define the collection coordinates, and how a user would refer to collection contents. Will use `namespace: confluent` and `name: platform`. (https://github.com/confluentinc/cp-ansible/pull/623). To use a role, using Kafka Broker as an example, this could look like this:
    ```yaml
    roles:
      - role: confluent.kafka.kafka_broker
    ```
- [x] Move/refactor the test scenarios out to a top level molecule directory. I do not think they should be mixed into the collection roles. (https://github.com/confluentinc/cp-ansible/pull/701)
- [x] Rename roles to "simple" names. 'The character `.` is not allowed in collection role names. And the collection will namespace the roles anyway. (https://github.com/confluentinc/cp-ansible/pull/624)
- [x] Move playbooks with tasks directory to a new playbooks directory. (https://github.com/confluentinc/cp-ansible/pull/730)
- [x] Move/refactor `roles/confluent.variables/filter_plugins` to `plugins/filter` (collection roles are not allowed to contain plugins) (https://github.com/confluentinc/cp-ansible/pull/631)
- [x] Update relative imports work in collections to start with a period (`.`). This is detailed in 3. [here](https://docs.ansible.com/ansible/latest/dev_guide/migrating_roles.html#migrating-other-role-plugins-to-a-collection). (https://github.com/confluentinc/cp-ansible/pull/709)
- [x] Convert molecule scenarios to test collections. Jeff Geerling has written an [excellent blog post](https://www.jeffgeerling.com/blog/2019/how-add-integration-tests-ansible-collection-molecule) about adding molecule integration tests to Ansible collections. (https://github.com/confluentinc/cp-ansible/pull/631)
- [x] Update Dev Guide, calling out that we are using Ansible collections, and all custom plugins must use FQPN (https://github.com/confluentinc/cp-ansible/pull/729)
- [x] Move test roles into own roles dir. We do not want to "ship" our testing roles with the collection. (https://github.com/confluentinc/cp-ansible/pull/734)

The Ansible dev documentation on [migrating roles to collections](https://docs.ansible.com/ansible/latest/dev_guide/migrating_roles.html#migrating-a-role-with-plugins-to-a-collection) might be useful for this migration.

I would be happy to help with this migration, but I think it should be coordinated by maintainers. Rebasing a refactor PR branch can be very annoying, and we must also ensure that any fixes done in cp-ansible are not "lost in refactoring". And personally, I would also say that fixing https://github.com/confluentinc/cp-ansible/issues/551 is a requirement for heavy contribution...

### PR DESCRIPTION
**Pull request by erikgb**
# Description

This PR restructures this project to be in accordance to the Ansible Collection requirements, and allows the playbooks and roles to be used in a bigger Ansible play - and avoid the pesky Git submodules. More description of the changes in the issue/epic: #613.

This change would also be a big step towards resolving https://github.com/confluentinc/cp-ansible/issues/113, but that will have to be fixed in a follow-up PR.

**This is a breaking change with regards to the user interface**, and we should also probably considering it generally breaking - since we do not know exactly how users use cp-ansible. **A documentation update is required**.

Fixes #613

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: meta/runtime.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+requires_ansible: ">=2.9"
```
Comment (domenicbove) [inline meta/runtime.yml:2]: 2.11 for the playbook feature: https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-a-playbook-from-a-collection

File: roles/common/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (domenicbove) [inline roles/common/meta/main.yml:1]: is this a new file, or a changed name file?

File: meta/runtime.yml (hunk 1 above)
Comment (erikgb) [reply meta/runtime.yml:2]: I'll fix it. But let's see a sucessful build first.

File: roles/common/meta/main.yml (hunk 2 above)
Comment (domenicbove) [reply roles/common/meta/main.yml:1]: i think the issue is this file needs to be moved/ edited in separate commits: https://stackoverflow.com/questions/433111/how-to-make-git-mark-a-deleted-and-a-new-file-as-a-file-move/433114

Comment (erikgb) [reply roles/common/meta/main.yml:1]: It is moved - from here: https://github.com/confluentinc/cp-ansible/blob/6.2.0-post/roles/confluent.common/meta/main.yml

Comment (erikgb) [reply roles/common/meta/main.yml:1]: It is marked as moved in the commit: https://github.com/confluentinc/cp-ansible/pull/698/commits/ed2b43a9f5223d8409fa81975a68de878e376370

I think you have to look at the individual commits, and not the PR diff.

Comment (domenicbove) [reply roles/common/meta/main.yml:1]: ok great! thanks for confirming

File: meta/runtime.yml (hunk 1 above)
Comment (erikgb) [reply meta/runtime.yml]: Changed the requirement to ansible-core >= 2.11

File: galaxy.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+namespace: confluent
+name: platform
+version: 6.2.0
```
Comment (domenicbove) [inline galaxy.yml:4]: this variable will need to be updated for the 7.0.0 release... wonder if theres a way to make it easier

Comment (domenicbove) [reply galaxy.yml:4]: there is a bumpconfig file that we can update with this

File: roles/confluent.common/meta/main.yml [hunk 4]
```diff
@@ -1,3 +0,0 @@
----
```
Comment (domenicbove) [inline roles/confluent.common/meta/main.yml:1]: I forget if this is a github thing, but shouldnt we see these files as moved, and a line changed... i guess its not a big deal. this file does not get updated ever so merge conflicts wont really happen

**Review by domenicbove — APPROVED** [review]
Wow so many files! Looks good. I think we could consider the .bumpconfig file for the version in the galaxy.yml, but its not currently in our dev guide so its not a process currently... meaning it shouldnt hold this up

File: roles/confluent.common/meta/main.yml (hunk 4 above)
Comment (erikgb) [reply roles/confluent.common/meta/main.yml:1]: It IS a GitHub thing, when you look at the total of the PR. But it you look at the commits, the history is preserved:

````
D:~/projects/github/confluent-cp-ansible (70-collections) $ git --no-pager log --follow roles/common/meta/main.yml
commit 3ac2b95ee3c67abee51ecfde19563c24351797bc
Author: Erik Godding Boye <egboye@gmail.com>
Date:   Thu Apr 1 21:30:37 2021 +0200

    Rename confluent.common role

commit def2fa498f89e2d50024d50efcb286fd9d09493c
Author: Erik Godding Boye <erik.boye@statnett.no>
Date:   Thu Apr 1 11:09:29 2021 +0200

    Rename confluent.variables role

commit ecbe0293a7c8cc2e1146ddb404edcc8df22d442e
Author: Domenic Bove <domenicbove@gmail.com>
Date:   Thu Jun 25 10:05:17 2020 -0700

    renamed confluent.variables_handlers to confluent.variables

commit 75c47ab41a5e0e0e8d8c54dfaca167cac37c5dbe
Author: Domenic Bove <domenicbove@gmail.com>
Date:   Mon Sep 16 10:19:13 2019 -0700

    renamed confluent.variables to confluent.variables_handlers

commit 45cb20f0a64d66ee666c4e0b98405146ff3a80cf
Author: Domenic Bove <domenicbove@gmail.com>
Date:   Fri Sep 13 11:53:36 2019 -0700

    default var for jolokia enabled created passed to all roles, need to build logic around empty strings

commit 230cd424caa4c2bd321bb637082b11cc986d49df
Author: Domenic Bove <domenicbove@gmail.com>
Date:   Tue Sep 3 14:39:11 2019 -0700

    roles now depend on vars role
D:~/projects/github/confluent-cp-ansible (70-collections) $
````

So I am quite sure this is not a problem.

File: galaxy.yml (hunk 3 above)
Comment (erikgb) [reply galaxy.yml:4]: If this can be improved, I will happily merge a PR to fix this into this PR. 😉

**Comment by domenicbove** [comment]
Just ran our "test harness" against this branch, it does not always run perfectly, some scenarios dont finish. These are the scenarios I got failures from:

Didn't finish:
- archive-plain-rhel
- mtls-custombundle-rhel
- zookeeper-mtls


- ccloud - looks like rel path issue *need to fix on this branch*
```
[31mfatal: [kafka-connect1]: FAILED! => {"changed": false, "msg": "Could not find or access '../../generated_ssl_files/snakeoil-ca-1.crt'\nSearched in:\n\t/home/centos/repos/ansible_collections/confluent/platform/molecule/ccloud/files/../../generated_ssl_files/snakeoil-ca-1.crt\n\t/home/centos/repos/ansible_collections/confluent/platform/molecule/ccloud/../../generated_ssl_files/snakeoil-ca-1.crt\n\t/home/centos/repos/ansible_collections/confluent/platform/molecule/ccloud/files/../../generated_ssl_files/snakeoil-ca-1.crt\n\t/home/centos/repos/ansible_collections/confluent/platform/molecule/ccloud/../../generated_ssl_files/snakeoil-ca-1.crt on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}[0m
```

- kafka-connect-replicator-mtls-scram-rhel
```
[31mfatal: [kafka-broker1]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": true, "cmd": "kafka-console-consumer --topic test-replicator-source --bootstrap-server kafka-broker1:9092 --from-beginning --timeout-ms 10000 --consumer.config /etc/kafka/client.properties\n", "delta": "0:00:01.571938", "end": "2021-08-05 17:06:52.135575", "failed_when_result": true, "msg": "non-zero return code", "rc": 1, "start": "2021-08-05 17:06:50.563637", "stderr": "Exception in thread \"main\" java.nio.file.NoSuchFileException: /etc/kafka/client.properties\n\tat java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)\n\tat java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)\n\tat java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)\n\tat java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)\n\tat java.base/java.nio.file.Files.newByteChannel(Files.java:371)\n\tat java.base/java.nio.file.Files.newByteChannel(Files.java:422)\n\tat java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)\n\tat java.base/java.nio.file.Files.newInputStream(Files.java:156)\n\tat org.apache.kafka.common.utils.Utils.loadProps(Utils.java:629)\n\tat org.apache.kafka.common.utils.Utils.loadProps(Utils.java:616)\n\tat kafka.tools.ConsoleConsumer$ConsumerConfig.<init>(ConsoleConsumer.scala:295)\n\tat kafka.tools.ConsoleConsumer$.main(ConsoleConsumer.scala:51)\n\tat kafka.tools.ConsoleConsumer.main(ConsoleConsumer.scala)", "stderr_lines": ["Exception in thread \"main\" java.nio.file.NoSuchFileException: /etc/kafka/client.properties", "\tat java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)", "\tat java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)", "\tat java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)", "\tat java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)", "\tat java.base/java.nio.file.Files.newByteChannel(Files.java:371)", "\tat java.base/java.nio.file.Files.newByteChannel(Files.java:422)", "\tat java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)", "\tat java.base/java.nio.file.Files.newInputStream(Files.java:156)", "\tat org.apache.kafka.common.utils.Utils.loadProps(Utils.java:629)", "\tat org.apache.kafka.common.utils.Utils.loadProps(Utils.java:616)", "\tat kafka.tools.ConsoleConsumer$ConsumerConfig.<init>(ConsoleConsumer.scala:295)", "\tat kafka.tools.ConsoleConsumer$.main(ConsoleConsumer.scala:51)", "\tat kafka.tools.ConsoleConsumer.main(ConsoleConsumer.scala)"], "stdout": "", "stdout_lines": []}[0m
```

- kafka-connect-replicator-plain-kerberos-rhel - rel path issue
```
[31mfatal: [mds-kafka-broker1]: FAILED! => {"changed": false, "msg": "Could not find or access 'roles/confluent.test/molecule/kafka-connect-replicator-plain-kerberos-rhel/generated_ssl_files/public.pem'\nSearched in:\n\t/home/centos/repos/ansible_collections/confluent/platform/roles/kafka_broker/files/roles/confluent.test/molecule/kafka-connect-replicator-plain-kerberos-rhel/generated_ssl_files/public.pem\n\t/home/centos/repos/ansible_collections/confluent/platform/roles/kafka_broker/roles/confluent.test/molecule/kafka-connect-replicator-plain-kerberos-rhel/generated_ssl_files/public.pem\n\t/home/centos/repos/ansible_collections/confluent/platform/roles/kafka_broker/tasks/files/roles/confluent.test/molecule/kafka-connect-replicator-plain-kerberos-rhel/generated_ssl_files/public.pem\n\t/home/centos/repos/ansible_collections/confluent/platform/roles/kafka_broker/tasks/roles/confluent.test/molecule/kafka-connect-replicator-plain-kerberos-rhel/generated_ssl_files/public.pem\n\t/home/centos/repos/ansible_collections/confluent/platform/playbooks/./files/roles/confluent.test/molecule/kafka-connect-replicator-plain-kerberos-rhel/generated_ssl_files/public.pem\n\t/home/centos/repos/ansible_collections/confluent/platform/playbooks/./roles/confluent.test/molecule/kafka-connect-replicator-plain-kerberos-rhel/generated_ssl_files/public.pem on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}[0m
```

Issue with debian dockerfile
- mtls-debian
- rbac-kerberos-debian
- rbac-mds-kerberos-debian
- rbac-plain-provided-debian
```
TASK [Build an Ansible compatible image (new)] *********************************
[1;30mFAILED - RETRYING: Build an Ansible compatible image (new) (3 retries left).[0m
```


- multi-ksql-connect-rhel - failure in verify tasks
```
TASK [Get Connectors on connect cluster2] **************************************
[31mfatal: [kafka-connect3]: FAILED! => {"changed": false, "elapsed": 0, "msg": "Status code was -1 and not [200]: Request failed: <urlopen error [SSL: SSLV3_ALERT_BAD_CERTIFICATE] sslv3 alert bad certificate (_ssl.c:618)>", "redirected": false, "status": -1, "url": "https://kafka-connect3:8083/connectors"}[0m
```

- rbac-mds-mtls-custom-rhel
```
TASK [confluent.platform.kafka_broker : Grant Audit Logs Principal ResourceOwner on confluent-audit-log-events Prefixed Topics] ***
[31mfatal: [mds-kafka-broker1]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'audit_logs_destintation_kafka_cluster_name' is undefined\n\nThe error appears to be in '/home/centos/repos/ansible_collections/confluent/platform/roles/kafka_broker/tasks/rbac_rolebindings.yml': line 32, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n# confluent iam rolebinding create --principal User:<audit-log-admin> --role ResourceOwner --resource Topic:confluent-audit-log-events --prefix --cluster-name audit_logs\n- name: Grant Audit Logs Principal ResourceOwner on confluent-audit-log-events Prefixed Topics\n  ^ here\n"}[0m
```

- rbac-mtls-rhel - same connect mtls error
```
TASK [Get Connect Cluster Details] *********************************************
[31mfatal: [kafka-connect1]: FAILED! => {"changed": false, "elapsed": 0, "msg": "Status code was -1 and not [200]: Request failed: <urlopen error [SSL: SSLV3_ALERT_BAD_CERTIFICATE] sslv3 alert bad certificate (_ssl.c:618)>", "redirected": false, "status": -1, "url": "https://localhost:8083"}[0m
```

**Review by domenicbove — APPROVED** [review]
LGTM

**Comment by domenicbove** [comment]
Resolved ccloud failure on this branch and kafka-connect-replicator-plain-kerberos-rhel failure on the 7.0.x branch

**Comment by domenicbove** [comment]
Resolved kafka-connect-replicator-mtls-scram-rhel failure on 7.0.x branch

**Comment by JumaX** [comment]
@erikgb @domenicbove I just pushed a few security fixes which were a hard requirement for this release.  Recommend pulling those into this PR, and doing the merge of this.  I am going hold off on any further changes until this is in, and will start reviewing today.

**Comment by erikgb** [comment]
> @erikgb @domenicbove I just pushed a few security fixes which were a hard requirement for this release. Recommend pulling those into this PR, and doing the merge of this. I am going hold off on any further changes until this is in, and will start reviewing today.

@JumaX Can you confirm that all your changes are now merged into the `7.0.x` branch? I will rebase this branch on top of it - as I have done multiple times already.

**Comment by JumaX** [comment]
@erikgb As mentioned all critical security updates are now merged.

**Comment by erikgb** [comment]
@JumaX I have just rebased this branch on top of the updated 7.0.x branch, so it should be ready for review now.

**Comment by JumaX** [comment]
@erikgb Great work.  I've been running a few molecule scenarios to validate functionality and I believe their is a common bug in all scenarios which use the prepare stage to import all.yml. It looks like the pathing is not correct.  Do you want to go ahead and correct this or would you prefer me to make a PR into this PR to correct it?

`rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel

Could not find or access '/Users/justin/all.yml' on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option
CRITICAL Ansible return code was 1, command was: ansible-playbook --inventory /Users/justin/.cache/molecule/cp-ansible/rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel/inventory --skip-tags molecule-notest,notest /Users/justin/repos/cp-ansible/molecule/rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel/prepare.yml
[justin@Justin-Manchester-MBP16:  cp-ansible] (70-collections)$ 

rbac-mds-mtls-custom-rhel

ERROR! Unable to retrieve file contents
Could not find or access '/Users/justin/repos/cp-ansible/molecule/rbac-mds-mtls-custom-rhel/confluent.platform.all' on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option
CRITICAL Ansible return code was 1, command was: ansible-playbook --inventory /Users/justin/.cache/molecule/cp-ansible/rbac-mds-mtls-custom-rhel/inventory --skip-tags molecule-notest,notest /Users/justin/repos/cp-ansible/molecule/rbac-mds-mtls-custom-rhel/prepare.yml

rbac-mtls-provided-ubuntu

ERROR! Unable to retrieve file contents
Could not find or access '/Users/justin/repos/cp-ansible/molecule/confluent.platform.all' on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option
CRITICAL Ansible return code was 1, command was: ansible-playbook --inventory /Users/justin/.cache/molecule/cp-ansible/rbac-mtls-provided-ubuntu/inventory --skip-tags molecule-notest,notest /Users/justin/repos/cp-ansible/molecule/collections_converge.yml
`

**Comment by erikgb** [comment]
@JumaX Did you follow the updated guidelines (in this PR) for cloning cp-ansible? https://github.com/confluentinc/cp-ansible/blob/70-collections/HOW_TO_TEST.md#cloning-cp-ansible

**Comment by JumaX** [comment]
@erikgb I cloned the 70-collections branch which fails with those errors.  

I have also followed the dev guide and created a brand new local repo and still run into the same problem.  If you look at the prepare.yml in those scenarios you will see that the pathing to all.yml is incorrect.

From the new repo test:

`ERROR! Unable to retrieve file contents
Could not find or access '/Users/justin/repos/ansible_collections/confluent/platform/molecule/rbac-mds-mtls-custom-rhel/confluent.platform.all' on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option
CRITICAL Ansible return code was 1, command was: ansible-playbook --inventory /Users/justin/.cache/molecule/platform/rbac-mds-mtls-custom-rhel/inventory --skip-tags molecule-notest,notest /Users/justin/repos/ansible_collections/confluent/platform/molecule/rbac-mds-mtls-custom-rhel/prepare.yml`

**Comment by erikgb** [comment]
@JumaX Yes, please check the versions of Ansible and Molecule! Consuming playbooks from Ansible collections are described here: https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-a-playbook-from-a-collection.

**Comment by JumaX** [comment]
@erikgb good call on the version, however it still looks like the pathing is broken on some of these scenarios.

`ansible [core 2.11.3]`

`[WARNING]: running playbook inside collection confluent.platform
ERROR! Unable to retrieve file contents
Could not find or access '/Users/justin/repos/ansible_collections/all.yml' on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option
CRITICAL Ansible return code was 1, command was: ansible-playbook --inventory /Users/justin/.cache/molecule/platform/rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel/inventory --skip-tags molecule-notest,notest /Users/justin/repos/ansible_collections/confluent/platform/molecule/rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel/prepare.yml`

**Comment by erikgb** [comment]
> `[WARNING]: running playbook inside collection confluent.platform ERROR! Unable to retrieve file contents Could not find or access '/Users/justin/repos/ansible_collections/all.yml' on the Ansible Controller. If you are using a module and expect the file to exist on the remote, see the remote_src option CRITICAL Ansible return code was 1, command was: ansible-playbook --inventory /Users/justin/.cache/molecule/platform/rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel/inventory --skip-tags molecule-notest,notest /Users/justin/repos/ansible_collections/confluent/platform/molecule/rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel/prepare.yml`

@JumaX Yes, it looks like some of the paths are broken. I will take a look tonight.

**Comment by erikgb** [comment]
> it still looks like the pathing is broken on some of these scenarios.

@JumaX I think the "paths" were just broken on the `rbac-kafka-connect-replicator-kerberos-mtls-custom-rhel` scenario. And that is because it seems to be a new scenario which was not yet added to the 7.0.x branch when we fixed the playbook references. But I have fixed it now. Good catch, and PTAL! 😄

**Review by JumaX — APPROVED** [review]
@erikgb amazing work!

**Comment by erikgb** [comment]
@JumaX Thx! I'll leave up to you and the rest of the team when to merge this. 😄

**Pull request merged**